src/features/filter.ts

```diff
@@ -1,12 +1,25 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { PayloadAction, Slice, createSlice } from '@reduxjs/toolkit';
+import { Status } from '../types/Status';
+
+type FilterState = {
+  query: string;
+  status: Status;
+};
 
 const initialState = {
   query: '',
-  status: 'all',
+  status: Status.All,
 };
 
-export const filterSlice = createSlice({
+export const filterSlice: Slice<FilterState> = createSlice({
   name: 'filter',
   initialState,
-  reducers: {},
+  reducers: {
+    setFilter: (state, { payload }: PayloadAction<Status>) => {
+      return { ...state, status: payload };
```

Comment:
```suggestion
      state.query = action.payload;
```
By the way, Redux Toolkit's createSlice uses Immer under the hood, which allows you to mutate the state directly without needing to return a new object.